### ISSUE TITLE
Sprites appear and disappear randomly with GLES3.

### ISSUE THREAD

**Issue by Bucham** [godotengine/godot#33804]
**Godot version:**
Godot 3.1.2
**OS/device including version:**
Windows 10/ HP laptop. Intel Core i7_8565U with UHD 620 integrated card
**Issue description:**
In a 2D project. The sprites appear and disappear randomly, when they intersect with others that are in the same z-index. The same does not occur in GLS2. 
**Steps to reproduce:**
Place several sprites that intersect in the same z index.

**Comment by volzhs** [comment]
without a reproducible sample project, it could be really hard to figure out what was wrong.

**Comment by clayjohn** [comment]
Can you upload an example project so that others can try to reproduce your issue?

**Comment by Bucham** [comment]
First of all thank you very much for your speed.
While trying to reproduce the problem to upload it, I discovered where the problem is. Although now I do not understand anything.
It turns out that everything comes by a character light, but it is a normal and ordinary light. It's very weird. I upload a video for you to see and judge on your own. It only happens with that particular light. It's very weird. In principle I would rather not have to upload the project, if necessary I would prefer to send it to someone in particular. What do you think about the problem? Thank you very much again. [https://anonfile.com/0e47x4Ben6/2019-11-22_01-05-32_mkv](url)

**Comment by clayjohn** [comment]
What we prefer is if you create a new project with the smallest number of nodes needed in order to reproduce the bug.

**Comment by volzhs** [comment]
Your full project could be also difficult to find what the problem is. :smile: 
I couldn't find a clue with your video.

**Comment by Bucham** [comment]
Ok, I got it, I had to erase all that doesn't contribute from my project because I couldn't replicate it. But that's it. Only 5 or six nodes and a new light. You can see the problem playing the file, or simply moving the light over the objects or the stage in the editor. You're going to kill me, but I don't control this page at all. Would you mind telling me, how I can upload the project?

**Comment by clayjohn** [comment]
Create a .zip file and then drag and drop it into the comment box here on GitHub. :)

**Comment by Bucham** [comment]
Hello again, I have got more information. I have managed to replicate the problem in a new project in the simplest way. Simply create a new node as a world. Put some sprite, and then a camera and a light. All as children of the world. If you place the camera as the first child, the light disappears everything that is below the camera, when the light passes over. If you place the camera in the middle, the light only disappears the nodes that are below, and if you place the camera at the end, everything works normally.

**Comment by Bucham** [comment]
[Light_and_camera_problem.zip](https://github.com/godotengine/godot/files/3879389/Light_and_camera_problem.zip)

**Comment by Zireael07** [comment]
If it's caused by the light2D, then it's a duplicate of an already reported problem.

**Comment by Bucham** [comment]
ah, thanks, I've tried it in 3.1 stable and the same thing happens. Thank you so much for your attention.

**Comment by Ranoller** [comment]
I confirm this bug in GLES3 in a laptop with a node from dragon bones plugin. With GLES3 renderer and zoom different than 0, some parts of the bodies dissapear. Same doesn´t happend in GLES2. At the same time, editor lags in GLES3, nor in GLES2.

**Comment by Bucham** [comment]
Hello again I have downloaded the stable version of Godot 3.2. The problem with the camera and lights has improved or at least changed. The sample project is configured as follows. It is a 2d project, with four sprites, a camera and a light.
What has changed is that now, when you play, everything works correctly, but not in the editor. The following occurs in the editor. The placement of the light in the node stack does not matter, what affects is the position of the camera. If you place it before all the sprites, the first sprite does not light up, and the others disappear.
If you place it after the first sprite, the second sprite does not light and the third and fourth disappear. So on. If you place it between the third and fourth sprite, the fourth sprite simply does not light up. If you put the camera at the end, everything works as it should. Basically what has changed is that now, when you play the game, everything seems to work fine (except in the editor) and that the sprite on which the light is put on does not disappear, but simply does not light up, only the sprites below. Here I leave the project in case you are interested. Thank you very much and greetings.
[SpritesDisappear.zip](https://github.com/godotengine/godot/files/4141354/SpritesDisappear.zip)

**Comment by Bucham** [comment]
1207/5000
Hello again I have downloaded the stable version of Godot 3.2. The problem with the camera and lights has improved or at least changed. The sample project is configured as follows. It is a 2d project, with four sprites, a camera and a light.
What has changed, is that now, when you play, everything works correctly, but not in the editor. The following occurs in the editor. The placement of the light in the stack of nodes does not matter, what affects is the position of the camera. If you place it before all the sprites, the first sprite does not light up, and the others disappear.
If you place it after the first sprite, the second sprite does not light and the third and fourth disappear. So on. If you place it between the third and fourth sprite, the fourth sprite simply does not light up. If you put the camera at the end, everything works as it should. Basically what has changed is that now, when you play the game, everything seems to work fine (except in the editor) and that the sprite on which the light is put on does not disappear, but simply does not light up, only the sprites below. Here I leave the project in case you are interested. Thank you very much and greetings.
[SpritesDisappears.zip](https://github.com/godotengine/godot/files/4141434/SpritesDisappears.zip)

**Comment by Bucham** [comment]
1207/5000
Hello again I have downloaded the stable version of Godot 3.2. The problem with the camera and lights has improved or at least changed. The sample project is configured as follows. It is a 2d project, with four sprites, a camera and a light.
What has changed is that now, when you play, everything works correctly, but not in the editor. The following occurs in the editor. The placement of the light in the node stack does not matter, what affects is the position of the camera. If you place it before all the sprites, the first sprite does not light up, and the others disappear.
If you place it after the first sprite, the second sprite does not light and the third and fourth disappear. So on. If you place it between the third and fourth sprite, the fourth sprite simply does not light up. If you put the camera at the end, everything works as it should. Basically what has changed is that now, when you play the game, everything seems to work fine, except in the editor, where the sprite on which the light is put on does not disappear, but simply does not light up, only the sprites disappear from below. Here I leave the project in case you are interested. Thank you very much and greetings.
[SpritesDisappears.zip](https://github.com/godotengine/godot/files/4141458/SpritesDisappears.zip)

**Comment by Bucham** [comment]
Hello again I have downloaded the stable version of Godot 3.2. The problem with the camera and lights has improved or at least changed. The sample project is configured as follows. It is a 2d project, with four sprites, a camera and a light.
What has changed, is that now, when you play, everything works correctly, but not in the editor. The following occurs in the editor. The placement of the light in the stack of nodes does not matter, what affects is the position of the camera. If you place it before all the sprites and place the light on the first sprite, the first sprite does not light up, and the others disappear.
If you place it after the first sprite and place the light on the second sprite, the second sprite does not light up and the third and fourth disappear. So on. If you place it between the third and fourth sprite, the fourth sprite simply does not light up. If you put the camera at the end, everything works as it should. Basically what has changed is that now, when you play the play, everything seems to work fine, except in the editor, where the sprite on which the light is put on does not disappear as before, but simply does not light up, they just disappear The sprites below. Here I leave the project in case you are interested. Thank you very much and greetings.
[SpritesDisappears.zip](https://github.com/godotengine/godot/files/4141493/SpritesDisappears.zip)

**Comment by clayjohn** [comment]
@Bucham I have run your test project and I cannot reproduce on Ubuntu 16.04 Mesa drivers Intel integrated graphics. :/

Have you tried updating your graphics drivers?

**Comment by Bucham** [comment]
Hi, I bought my computer a few months ago and updated everything that HP support allows me. It seems that being a laptop does not allow me to update the drivers freely, only with their updates. I tried to update them on my own at first but they didn't allow me. So no, I think they are not the last. It seems to me that I won't be able to do it until those of HP decide to update

**Comment by clayjohn** [comment]
@Bucham That is very unfortunate. To me, this sounds like a driver issue. 

I have a windows 10 computer at home, I will try on that and see if its a windows issue.

**Comment by Bucham** [comment]
Ok, thank you very much. Tomorrow I will test it on my father's computer, with an updated amd card, to see what happens, and I give you more information.

**Comment by Bucham** [comment]
Hi. I have already tried it on my father's computer with windows 10 and a geforce 1030 and it works correctly. My current driver is x.x.x.6471 and intel says the last one available is x.x.x.7755. I tried to update it again, but hp does not allow me to install it. I do not know if my problem will happen to all users with an hp laptop and uhd 620 integrated card (my computer is not even 5 months old, it is ridiculous that they do not allow you to freely update the drivers, I have the same driver problem with godot  in my old surface pro6). I guess the problem is already clear. Thank you very much for your attention anyway.
